pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/forward/VarByteChunkForwardIndexReaderV4.java

```diff
@@ -113,6 +116,156 @@ public byte[] getBytes(int docId, ReaderContext context) {
+    }
+    return valueBuffer;
+  }
+
+  @Override
+  public int getLongMV(int docId, long[] valueBuffer, VarByteChunkForwardIndexReaderV4.ReaderContext context) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(context.getValue(docId));
+    int numValues = byteBuffer.getInt();
+    for (int i = 0; i < numValues; i++) {
+      valueBuffer[i] = byteBuffer.getLong();
+    }
+    return numValues;
+  }
+
+  @Override
+  public long[] getLongMV(int docId, VarByteChunkForwardIndexReaderV4.ReaderContext context) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(context.getValue(docId));
+    int numValues = byteBuffer.getInt();
+    long[] valueBuffer = new long[numValues];
+    for (int i = 0; i < numValues; i++) {
+      valueBuffer[i] = byteBuffer.getLong();
+    }
+    return valueBuffer;
+  }
+
+  @Override
+  public int getFloatMV(int docId, float[] valueBuffer, VarByteChunkForwardIndexReaderV4.ReaderContext context) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(context.getValue(docId));
+    int numValues = byteBuffer.getInt();
+    for (int i = 0; i < numValues; i++) {
+      valueBuffer[i] = byteBuffer.getFloat();
+    }
+    return numValues;
+  }
+
+  @Override
+  public float[] getFloatMV(int docId, VarByteChunkForwardIndexReaderV4.ReaderContext context) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(context.getValue(docId));
+    int numValues = byteBuffer.getInt();
+    float[] valueBuffer = new float[numValues];
+    for (int i = 0; i < numValues; i++) {
+      valueBuffer[i] = byteBuffer.getFloat();
+    }
+    return valueBuffer;
+  }
+
+  @Override
+  public int getDoubleMV(int docId, double[] valueBuffer, VarByteChunkForwardIndexReaderV4.ReaderContext context) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(context.getValue(docId));
+    int numValues = byteBuffer.getInt();
+    for (int i = 0; i < numValues; i++) {
+      valueBuffer[i] = byteBuffer.getDouble();
+    }
+    return numValues;
+  }
+
+  @Override
+  public double[] getDoubleMV(int docId, VarByteChunkForwardIndexReaderV4.ReaderContext context) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(context.getValue(docId));
+    int numValues = byteBuffer.getInt();
+    double[] valueBuffer = new double[numValues];
+    for (int i = 0; i < numValues; i++) {
+      valueBuffer[i] = byteBuffer.getFloat();
+    }
+    return valueBuffer;
+  }
+
+  @Override
+  public int getStringMV(int docId, String[] valueBuffer, VarByteChunkForwardIndexReaderV4.ReaderContext context) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(context.getValue(docId));
+    int numValues = byteBuffer.getInt();
+    int contentOffset = (numValues + 1) * Integer.BYTES;
+    for (int i = 0; i < numValues; i++) {
+      int length = byteBuffer.getInt((i + 1) * Integer.BYTES);
+      byte[] bytes = new byte[length];
+      byteBuffer.position(contentOffset);
+      byteBuffer.get(bytes, 0, length);
+      valueBuffer[i] = new String(bytes, StandardCharsets.UTF_8);
+      contentOffset += length;
```

Comment:
Do we need to maintain contentOffset? `position` it once before the for loop should be good